packages/@glimmer/compiler/lib/compiler.ts

```diff
@@ -53,25 +54,200 @@ const defaultOptions: PrecompileOptions = {
  *     let template = factory.create(env);
  *
  * @method precompile
- * @param {string} string a Glimmer template string
+ * @param {string} source a Glimmer template string
  * @return {string} a template javascript string
  */
-export function precompile(string: string, options?: PrecompileOptions): TemplateJavascript;
+export function precompile(source: string, options?: PrecompileOptions): TemplateJavascript;
 export function precompile(
-  string: string,
+  source: string,
   options: PrecompileOptions = defaultOptions
 ): TemplateJavascript {
-  let ast = preprocess(string, options);
+  let ast = preprocess(source, options);
   let { meta } = options;
-  let { block } = TemplateCompiler.compile(ast, string, options);
+  let { block } = TemplateCompiler.compile(ast, source, options);
+
+  let blockJSON = block.toJSON();
+  let blockJSONString = JSON.stringify(blockJSON);
+
   let idFn = options.id || defaultId;
-  let blockJSON = JSON.stringify(block.toJSON());
-  let templateJSONObject: SerializedTemplateWithLazyBlock<unknown> = {
-    id: idFn(JSON.stringify(meta) + blockJSON),
-    block: blockJSON,
-    meta,
-  };
-
-  // JSON is javascript
-  return JSON.stringify(templateJSONObject);
+  let id = idFn(JSON.stringify(meta) + blockJSONString);
+
+  let debug = new WireFormatDebugger(blockJSON);
+
+  let obj = new ObjectCompiler();
+
+  obj.set('id', id);
+  obj.set('meta', meta);
+
+  let func = new FunctionCompiler(obj.indentation);
+
+  func.push('/*');
+  func.push();
+
+  func.push('Source:');
+  func.push();
+  func.push('```hbs');
+  func.push(source);
```

Comment:
This source needs to be escaped, it can **absolutely** contain `*/` in it...

Consider compiling the following template:

```hbs
LOL */ you screwed now dude
```